src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -430,9 +338,7 @@ MavlinkReceiver::send_flight_information()
 	param_t param_flight_uuid = param_find("COM_FLIGHT_UUID");
 
 	if (param_flight_uuid != PARAM_INVALID) {
-		int32_t flight_uuid;
-		param_get(param_flight_uuid, &flight_uuid);
-		flight_info.flight_uuid = (uint64_t)flight_uuid;
+		flight_info.flight_uuid = static_cast<uint64_t>(_p_flight_uuid.get());
```

Comment:
Other than the PARAM_INVALID check I think it was fine before. I didn't see that the destination field in flight info was a uint64_t.
You should continue using the new parameter wrapper which "finds" the param at construction and marks it active in the system. Activating a param later in boot can cause issues during mavlink sync.

Reply:
@dagar, I've removed the `PARAM_INVALID` check, (caught at compile time now with the macro usage), and pushed that commit up.  Let me know if you'd like me to squash the commits from this PR into a single commit.